File: code/go/ch16/src/luago/vm/inst_misc.go
package vm

import . "luago/api"

// R(A) := R(B)
func move(i Instruction, vm LuaVM) {
	a, b, _ := i.ABC()
	a += 1
	b += 1

	vm.Copy(b, a)
}

// pc+=sBx; if (A) close all upvalues >= R(A - 1)
func jmp(i Instruction, vm LuaVM) {
	a, sBx := i.AsBx()

	vm.AddPC(sBx)
	if a != 0 {
		vm.CloseUpvalues(a)
	}
}

File: code/csharp/ch11/luavm/vm/InstMisc.cs
using LuaVM = luavm.state.LuaState;

namespace luavm.vm
{
    public static class InstMisc
    {
        internal static void move(Instruction i, ref LuaVM vm)
        {
            var ab_ = i.ABC();
            var a = ab_.Item1 + 1;
            var b = ab_.Item2 + 1;
            vm.Copy(b, a);
        }

        internal static void jmp(Instruction i, ref LuaVM vm)
        {
            var (a, sBx ) = i.AsBx();
            ;

            vm.AddPC(sBx);
            if (a != 0 )
            {
                vm.CloseUpvalues(a);
            }
        }
    }
}